File: src/main/java/br/com/blz/testjava/mapper/InventoryMapper.java
package br.com.blz.testjava.mapper;

import java.math.BigDecimal;

import br.com.blz.testjava.dto.InventoryDTO;
import br.com.blz.testjava.model.Inventory;

public class InventoryMapper {
	
	private InventoryMapper() {}
	
	public static Inventory toEntity(InventoryDTO inventoryDTO) {
		
		
		Inventory inventory = new Inventory();
		if(inventoryDTO == null) {
			return inventory;
		}
		inventory.setWareHouses(WareHousesMapper.toEntities(inventoryDTO.getWarehouses()));
		
		return inventory;
	}

	public static InventoryDTO toDTO(Inventory inventory) {
		
		InventoryDTO dto = new InventoryDTO();
		if(inventory == null) {
			return dto;
		}
		dto.setWarehouses(WareHousesMapper.toDTOS(inventory.getWareHouses()));
		dto.setQuantity(inventory.getWareHouses().stream()
				.map(w -> w.getQuantity())
				.reduce(BigDecimal.ZERO, BigDecimal::add));
		
		return dto;
	}

}

File: src/main/java/br/com/blz/testjava/inventory/mapper/InventoryMapper.kt
package br.com.blz.testjava.inventory.mapper

import br.com.blz.testjava.inventory.dto.InventoryInDTO
import br.com.blz.testjava.inventory.dto.InventoryOutDTO
import br.com.blz.testjava.inventory.model.Inventory
import br.com.blz.testjava.warehouse.mapper.WarehouseMapper

class InventoryMapper {
  companion object {

    fun toEntity(inventoryInDTO: InventoryInDTO): Inventory {
      return Inventory(inventoryInDTO.warehouses.map { WarehouseMapper.toEntity(it) })
    }

    fun toOutDto(inventory: Inventory): InventoryOutDTO {
      return InventoryOutDTO(
        inventory.quantity,
        inventory.warehouses.map { WarehouseMapper.toDto(it) })
    }
  }
}
